go/registry/api/api.go

```diff
@@ -448,7 +448,11 @@ func VerifyRegisterNodeArgs( // nolint: gocyclo
 			return nil, ErrInvalidArgument
 		}
 
-		if err := verifyAddresses(params, n.Consensus.Addresses); err != nil {
+		addresses := make([]node.Address, len(n.Consensus.Addresses))
+		for _, v := range n.Consensus.Addresses {
+			addresses = append(addresses, v.Address)
+		}
+		if err := verifyAddresses(params, addresses); err != nil {
```

Comment:
Shouldn't `verifyAddresses` be updated instead to also verify IDs (at least `.IsValid()`)?

Reply:
Yes, agreed.

I've extended `verifyAddresses()` to support checking both `[]node.ConsensusAddress` and `[]node.Address` types.